src/toil/test/lib/dockerTest.py

```diff
@@ -1,226 +1,309 @@
+                             deferParam=RM)
+
+    def testDockerClean_CxD_None(self):
+        self.testDockerClean(disableCaching=True, detached=True, rm=False,
+                             deferParam=None)
+
+    def testDockerClean_Cxx_FORGO(self):
+        self.testDockerClean(disableCaching=True, detached=False, rm=False,
+                             deferParam=FORGO)
+
+    def testDockerClean_Cxx_STOP(self):
+        self.testDockerClean(disableCaching=True, detached=False, rm=False,
+                             deferParam=STOP)
+
+    def testDockerClean_Cxx_RM(self):
+        self.testDockerClean(disableCaching=True, detached=False, rm=False,
+                             deferParam=RM)
+
+    def testDockerClean_Cxx_None(self):
+        self.testDockerClean(disableCaching=True, detached=False, rm=False,
+                             deferParam=None)
+
+    def testDockerClean_xRx_FORGO(self):
+        self.testDockerClean(disableCaching=False, detached=False, rm=True,
+                             deferParam=FORGO)
+
+    def testDockerClean_xRx_STOP(self):
+        self.testDockerClean(disableCaching=False, detached=False, rm=True,
+                             deferParam=STOP)
+
+    def testDockerClean_xRx_RM(self):
+        self.testDockerClean(disableCaching=False, detached=False, rm=True,
+                             deferParam=RM)
+
+    def testDockerClean_xRx_None(self):
+        self.testDockerClean(disableCaching=False, detached=False, rm=True,
+                             deferParam=None)
+
+    def testDockerClean_xxD_FORGO(self):
+        self.testDockerClean(disableCaching=False, detached=True, rm=False,
+                             deferParam=FORGO)
+
+    def testDockerClean_xxD_STOP(self):
+        self.testDockerClean(disableCaching=False, detached=True, rm=False,
+                             deferParam=STOP)
+
+    def testDockerClean_xxD_RM(self):
+        self.testDockerClean(disableCaching=False, detached=True, rm=False,
+                             deferParam=RM)
+
+    def testDockerClean_xxD_None(self):
+        self.testDockerClean(disableCaching=False, detached=True, rm=False,
+                             deferParam=None)
+
+    def testDockerClean_xxx_FORGO(self):
+        self.testDockerClean(disableCaching=False, detached=False, rm=False,
+                             deferParam=FORGO)
+
+    def testDockerClean_xxx_STOP(self):
+        self.testDockerClean(disableCaching=False, detached=False, rm=False,
+                             deferParam=STOP)
+
+    def testDockerClean_xxx_RM(self):
+        self.testDockerClean(disableCaching=False, detached=False, rm=False,
+                             deferParam=RM)
+
+    def testDockerClean_xxx_None(self):
+        self.testDockerClean(disableCaching=False, detached=False, rm=False,
+                             deferParam=None)
+
+    def testDockerPipeChain(self, disableCaching=True):
```

Comment:
I like the way you were able to expand the tests to cover all the parameter options, nice!

Reply:
Thanks!  And thanks for taking the time to review!